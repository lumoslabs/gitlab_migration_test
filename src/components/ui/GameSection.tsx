import GameCard from '@components/ui/GameCard'
import { Container } from 'react-bootstrap'
import { css, StyleSheet } from 'aphrodite/no-important'
import commonStyles from '@styles/commonStyles'
import { GameConfig } from '@backend/services/ConfigService'

const GameSection = ({ games, onClick }: { games: GameConfig[], onClick: (slug: string) => void }): JSX.Element => {

  const gameCards = games.map((props): JSX.Element => {
    const { id: slug } = props
    const { title, brain_area: brainArea, banner_url: bannerUrl } = props.values

    return (
      <GameCard
        onClick={onClick}
        key={`game_card_${title}`}
        title={title}
        brainArea={brainArea}
        bannerUrl={bannerUrl}
        slug={slug}
      />
    )
  })

  const gameCardColumns = []
  for (let col = 0, i = 0; col < ((gameCards.length / 2)); col++) {
    gameCardColumns.push(
      <div className={css([commonStyles.flexRowAllCenter, styles.gridCol])}
        key={`game_col_${i}`}
      >
        {gameCards[i]}
        {gameCards[i + 1]}
      </div>
    )
    i += 2
  }

  return (
    <Container className={css([commonStyles.flexRowAllCenter, styles.gamesDiv])}>
      {gameCardColumns}
    </Container>
  )
}

const styles = StyleSheet.create({
  gamesDiv: {
    display: 'flex',
    alignItems: 'flex-start',
    marginTop: '10px'
  },
  gridCol: {
    display: 'flex',
    flexDirection: 'column',
    flex: '0 0 auto'
  }
})

export default GameSection
