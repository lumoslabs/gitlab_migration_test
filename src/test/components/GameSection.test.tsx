import React from 'react'
import GameSection from '@components/ui/GameSection'
import { render } from '@testing-library/react'
import gamesMock from '@test/mocks/games'

it('matches snapshot', () => {
  const { asFragment } = render(
    <GameSection games={gamesMock} onClick={(slug) => { }} />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})

//TODO: test onclick function
