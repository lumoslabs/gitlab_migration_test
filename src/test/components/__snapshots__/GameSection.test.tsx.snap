// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GameSection matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexRowAllCenter_1cn94ih-o_O-gamesDiv_1idp2zd container"
  >
    <div
      class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
    >
      <a>
        <div
          class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
        >
          <div
            class="flexColumnAllCenter_1vexgx1-o_O-card_2cv6tx card"
          >
            <img
              alt="Game icon"
              class="card-img-top cardImg_1inpevm"
              src="https://assets.nest.lumosity.com/frontend_assets/banner/color_match_header.png"
            />
            <div
              class="flexColumn_n8t8ll-o_O-cardBody_ly1891 card-body"
            >
              <p
                class="pageTitle_1a3u8g5-o_O-subTitle_rlrlr4"
              >
                FLEXIBILITY
              </p>
              <div
                class="title_1wry8v1 card-title h5"
              >
                Color Match
              </div>
            </div>
          </div>
        </div>
      </a>
      <a>
        <div
          class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
        >
          <div
            class="flexColumnAllCenter_1vexgx1-o_O-card_2cv6tx card"
          >
            <img
              alt="Game icon"
              class="card-img-top cardImg_1inpevm"
              src="https://assets.nest.lumosity.com/frontend_assets/banner/ebb_and_flow_header.png"
            />
            <div
              class="flexColumn_n8t8ll-o_O-cardBody_ly1891 card-body"
            >
              <p
                class="pageTitle_1a3u8g5-o_O-subTitle_rlrlr4"
              >
                FLEXIBILITY
              </p>
              <div
                class="title_1wry8v1 card-title h5"
              >
                Ebb and Flow
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
    >
      <a>
        <div
          class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
        >
          <div
            class="flexColumnAllCenter_1vexgx1-o_O-card_2cv6tx card"
          >
            <img
              alt="Game icon"
              class="card-img-top cardImg_1inpevm"
              src="https://assets.nest.lumosity.com/frontend_assets/banner/raindrops_header.png"
            />
            <div
              class="flexColumn_n8t8ll-o_O-cardBody_ly1891 card-body"
            >
              <p
                class="pageTitle_1a3u8g5-o_O-subTitle_rlrlr4"
              >
                MATH
              </p>
              <div
                class="title_1wry8v1 card-title h5"
              >
                Raindrops
              </div>
            </div>
          </div>
        </div>
      </a>
      <a>
        <div
          class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
        >
          <div
            class="flexColumnAllCenter_1vexgx1-o_O-card_2cv6tx card"
          >
            <img
              alt="Game icon"
              class="card-img-top cardImg_1inpevm"
              src="https://assets.nest.lumosity.com/frontend_assets/banner/train_of_thought_header.png"
            />
            <div
              class="flexColumn_n8t8ll-o_O-cardBody_ly1891 card-body"
            >
              <p
                class="pageTitle_1a3u8g5-o_O-subTitle_rlrlr4"
              >
                ATTENTION
              </p>
              <div
                class="title_1wry8v1 card-title h5"
              >
                Train of Thought
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
    >
      <a>
        <div
          class="flexRowAllCenter_1cn94ih-o_O-gridCol_1f8v8h4"
        >
          <div
            class="flexColumnAllCenter_1vexgx1-o_O-card_2cv6tx card"
          >
            <img
              alt="Game icon"
              class="card-img-top cardImg_1inpevm"
              src="https://assets.nest.lumosity.com/frontend_assets/banner/word_snatchers_header.png"
            />
            <div
              class="flexColumn_n8t8ll-o_O-cardBody_ly1891 card-body"
            >
              <p
                class="pageTitle_1a3u8g5-o_O-subTitle_rlrlr4"
              >
                LANGUAGE
              </p>
              <div
                class="title_1wry8v1 card-title h5"
              >
                Word Snatchers
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
